src/wazuh_modules/syscollector/ports/portLinuxWrapper.h

```diff
@@ -0,0 +1,241 @@
+    { TCP_IPV6,                            TCP              }
+};
+
+static const std::map<PortType, IPVersion> IPVERSION_TYPE =
+{
+    { UDP_IPV4,                            IPV4             },
+    { UDP_IPV6,                            IPV6             },
+    { TCP_IPV4,                            IPV4             },
+    { TCP_IPV6,                            IPV6             }
+};
+
+static const std::map<int32_t, std::string> STATE_TYPE = 
+{
+    { TCP_ESTABLISHED,                     "established"    },
+    { TCP_SYN_SENT,                        "syn_sent"       },
+    { TCP_SYN_RECV,                        "syn_recv"       },
+    { TCP_FIN_WAIT1,                       "fin_wait1"      },
+    { TCP_FIN_WAIT2,                       "fin_wait2"      },
+    { TCP_TIME_WAIT,                       "time_wait"      },
+    { TCP_CLOSE,                           "close"          },
+    { TCP_CLOSE_WAIT,                      "close_wait"     },
+    { TCP_LAST_ACK,                        "last_ack"       },
+    { TCP_LISTEN,                          "listening"      },
+    { TCP_CLOSING,                         "closing"        }
+};
+
+
+class LinuxPortWrapper final : public IPortWrapper
+{
+    std::vector<std::string> m_fields;
+    PortType m_type;
+    std::vector<std::string> m_remoteAddress;
+    std::vector<std::string> m_localAddress;
+    std::vector<std::string> m_queue;
+    static std::string IPv4Address(const std::string& hexRawAddress) 
+    {
+        std::stringstream ss;
+        in_addr addr;
+        ss << std::hex << hexRawAddress;
+        ss >> addr.s_addr;
+        return inet_ntoa(addr);
+    }
+
+    static std::string IPv6Address(const std::string& hexRawAddress)
+    {
+        std::string retVal { "unknown" };
+        struct in6_addr sin6;
+        auto index { 0l };
+        const auto hexAddressLength { hexRawAddress.length() };
+        if (hexAddressLength == IPv6AddressHexSize)
+        {
+            char address[INET6_ADDRSTRLEN] = { 0 };
+            for (auto i = 0ull; i < hexAddressLength; i += CHAR_BIT)
+            {
+                std::stringstream ss;
+                ss << std::hex << hexRawAddress.substr(CHAR_BIT*index, CHAR_BIT);
+                ss >> sin6.s6_addr32[index];
+                ++index;
+            }
+            retVal = inet_ntop(AF_INET6, &sin6, address, sizeof(address));
+        }
+        return retVal;
+    }
+
+    public:
+    explicit LinuxPortWrapper(const PortType type, const std::string& row)
+    : m_fields{ Utils::split(row, ' ') }
+    , m_type { type }
+    , m_remoteAddress { std::move(Utils::split(m_fields.at(REMOTE_ADDRESS),':')) }
+    , m_localAddress { std::move(Utils::split(m_fields.at(LOCAL_ADDRESS),':')) }
+    , m_queue { std::move(Utils::split(m_fields.at(QUEUE),':')) }
+    { }
+
+    ~LinuxPortWrapper() = default;
+    std::string protocol() const override
+    {
+        return PORTS_TYPE.at(m_type);
```

Comment:
this will throw an exception is that the expected behavior? in all other cases it's being checked with iterator/find logic.